From 7a47fdb142ba73f2a0eccb5efd38bb13af075b73 Mon Sep 17 00:00:00 2001
From: Tao Huang <huangtao@rock-chips.com>
Date: Tue, 17 Sep 2019 15:34:50 +0800
Subject: [PATCH] Input: gt1x - enable asynchronous probing

It takes a bit of time to go through controller power up sequence and
initialization. To not stall the overall boot progress let's probe the
controller asynchronously, given that userspace is usually prepared for
hot-plugging of input devices and thus does not rely on particular
ordering.

Save about 0.35s on RK1808 EVB.

Change-Id: I5f38ae7dd9cbebeff02a5e522f29e131d21473ba
Signed-off-by: Tao Huang <huangtao@rock-chips.com>
---
 drivers/input/touchscreen/gt1x/gt1x.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/input/touchscreen/gt1x/gt1x.c b/drivers/input/touchscreen/gt1x/gt1x.c
index c14581113b92..60a79d8d2c02 100644
--- a/drivers/input/touchscreen/gt1x/gt1x.c
+++ b/drivers/input/touchscreen/gt1x/gt1x.c
@@ -739,6 +739,7 @@ static struct i2c_driver gt1x_ts_driver = {
 #if !defined(CONFIG_FB) && defined(CONFIG_PM)
 		   .pm = &gt1x_ts_pm_ops,
 #endif
+		   .probe_type = PROBE_PREFER_ASYNCHRONOUS,
 		   },
 };
 
-- 
2.35.3

